src/main/java/com/google/devtools/build/lib/analysis/BatchCommandConstructor.java

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.analysis;
+
+import com.google.common.collect.ImmutableList;
+import com.google.devtools.build.lib.actions.Artifact;
+import com.google.devtools.build.lib.analysis.actions.FileWriteAction;
+
+/**
+ * The class for constructing command line for Batch.
+ */
+public class BatchCommandConstructor implements CommandConstructor {
```

Comment:
"Batch" is an overloaded term.
Could we call it something else? How about `CmdExeCommandConstructor`?

Also in other places where it's now "batch".

Reply:
I thought of it as a batch-style CommandConstructor, in the sense of `bazel --batch`.

I actually don't know whether "Batch" is the name of the language. I don't know if this language even has a name... I know `.bat` files are called batch files, I guess because they batch commands together.

Reply:
Ah, I see. Indeed, this could be confusing. But it is commonly called "Windows Batch Script", see this link: https://en.wikibooks.org/wiki/Windows_Batch_Scripting

So to make it more clear, I think we can Rename it to "WindowsBatchCommandConstructor" and also for powershell "WindowsPowershellCommandConstructor"

Reply:
SGTM!

Reply:
DOne